### PR TITLE
Reuse postgres instance in tests

### PR DESCRIPTION
**Pull request by pm47**
Create new database for each unit test instead of spawning a new db instance.

### CODE REVIEW

**Comment by pm47** [comment]
I'm not too sure about this. The response time for db tests is significantly reduced, but overall build time is unchanged, and it adds complexity.

File: eclair-core/src/test/scala/fr/acinq/eclair/TestDatabases.scala [hunk 1]
```diff
@@ -112,13 +115,34 @@ object TestDatabases {
...
+      system.terminate()
+    }
     // @formatter:on
   }
 
+  object TestPgDatabases {
+    /** single instance */
+    private val pg = EmbeddedPostgres.start()
+
+    def getNewDatabase: DataSource = {
```
Comment (t-bast) [inline eclair-core/src/test/scala/fr/acinq/eclair/TestDatabases.scala:133]: nit: should look like a function call, not a getter on an internal field:
```suggestion
    def getNewDatabase(): DataSource = {
```

**Review by t-bast — COMMENTED** [review]
This isn't too much complexity and I believe it makes sense, it's worth having this :+1:

File: eclair-core/src/test/scala/fr/acinq/eclair/TestDatabases.scala [hunk 2]
```diff
@@ -95,13 +98,13 @@ object TestDatabases {
...
-    override def close(): Unit = ()
+    override def close(): Unit = connection.close()
     // @formatter:on
   }
 
   case class TestPgDatabases() extends TestDatabases {
-    private val pg = EmbeddedPostgres.start()
-    val datasource: DataSource = pg.getPostgresDatabase
+
+    val datasource: DataSource = getNewDatabase
```
Comment (t-bast) [inline eclair-core/src/test/scala/fr/acinq/eclair/TestDatabases.scala:107]: nit:
```suggestion
    val datasource: DataSource = getNewDatabase()
```

If we want to get rid of the warning that this looks like a getter (which is a dumb warning that only looks at the fact that the function name starts with `get`) we can rename that `createNewDatabase`

**Review by t-bast — APPROVED** [review]

**Pull request merged**